ddtrace/llmobs/_llmobs.py

```diff
@@ -75,6 +89,164 @@ def disable(cls) -> None:
+        return cls._instance._start_span("tool", name=name, session_id=session_id)
+
+    @classmethod
+    def task(cls, name: Optional[str] = None, session_id: Optional[str] = None) -> Span:
+        if cls.enabled is False or cls._instance is None:
+            raise Exception("LLMObs.task() cannot be used while LLMObs is disabled.")
+        return cls._instance._start_span("task", name=name, session_id=session_id)
+
+    @classmethod
+    def agent(cls, name: Optional[str] = None, session_id: Optional[str] = None) -> Span:
+        if cls.enabled is False or cls._instance is None:
+            raise Exception("LLMObs.agent() cannot be used while LLMObs is disabled.")
+        return cls._instance._start_span("agent", name=name, session_id=session_id)
+
+    @classmethod
+    def workflow(cls, name: Optional[str] = None, session_id: Optional[str] = None) -> Span:
+        if cls.enabled is False or cls._instance is None:
+            raise Exception("LLMObs.workflow() cannot be used while LLMObs is disabled.")
+        return cls._instance._start_span("workflow", name=name, session_id=session_id)
+
+    @classmethod
+    def annotate(
+        cls,
+        span: Optional[Span] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        input_data: Optional[Union[List[Dict[str, Any]], str]] = None,
+        output_data: Optional[Union[List[Dict[str, Any]], str]] = None,
+        tags: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Sets the parameter, input, and output tags for a given LLMObs span.
+        span: Span to annotate. If no span is provided, the current active span will be used.
+         Must be an LLMObs-type span.
+        parameters: Dictionary of input parameter key-value pairs such as max_tokens/temperature.
+         Will be mapped to span's meta.input.parameters.* fields.
+        input_data: A single input string, or a list of dictionaries following {"content": "...", "role": "..."}.
+         Will be mapped to `meta.input.value` or `meta.input.messages.*`, respectively.
+         For llm span types, string inputs will be wrapped in a message dictionary.
+        output_data: A single output string, or a list of dictionaries following {"content": "...", "role": "..."}.
+         Will be mapped to `meta.output.value` or `meta.output.messages.*`, respectively.
+         For llm span types, string outputs will be wrapped in a message dictionary.
+        tags: Dictionary of key-value pairs to tag the LLMObs span with.
+        """
+        if cls.enabled is False or cls._instance is None:
+            raise Exception("LLMObs.annotate() cannot be used while LLMObs is disabled.")
+        if span is None:
+            span = cls._instance.tracer.current_span()
+            if span is None:
+                raise ValueError("No span provided and no active span found.")
+        if span.span_type != SpanTypes.LLM:
+            raise ValueError("span must be an LLM-type span.")
```

Comment:
More of a curiosity question, def not blocking: is there a reason for raising exceptions over logging + gracefully returning? I know in the Node tracer we opt for the latter, in favor of a smooth user experience (except for instrumented library errors, which we make sure get thrown). 

I see you do a mix of both (as in the lines below), so I was just wondering if it's a best-practice Python thing, or something else. 

**Thought:** since this a new aspect of the user experience for the Python tracer, maybe logging could provide a better end experience?

Reply:
Good point, will think about logging rather than crashing the user app - we should aim to be as minimally disruptive as possible.